### DISCUSSION TITLE
Tracking face and body

### DISCUSSION THREAD

**Discussion by sevencong** [vladmandic/human#143] [Ideas]
Can we tracking detected face or/and body movement in multiple sequence frames in-order to GROUP detected faces. Then we do not need to match them later if we found in database.

example:
frame 1: [face group id1, face group id 2], -> bad quality, do .match() and not found in database
frame 2: [face group id1, face group id 2] -> bad quality, do .match() and not found in database
frame 3: [face group id1-nameX, face group id 2-nameY] -> GOOD quality, do .match()  and found in database
frame 4: [face group id1-nameX, face group id 2-nameY] -> not need to .match() again
frame 5: [face group id1-nameX, face group id 2-nameY] -> not need to .match() again

**Comment by vladmandic** [comment]
I'll think about it a bit more, but not sure how would that be any faster than using `match` explicitly as library would have to do the same thing anyhow.

**Reply by sevencong** [reply]
at some situations, .match() return low similarity on same face (different angle...), if we have them grouped (tracking), we have more information on faces matching.

**Reply by vladmandic** [reply]
question is how to group them as detection order depends on size and location within frame and that can change in video input.

**Reply by sevencong** [reply]
I found python code for tracking https://github.com/abewley/sort

**Reply by vladmandic** [reply]
I'll take a look
